### DISCUSSION TITLE
Strongswan Android client and AES hardware acceleration

### DISCUSSION THREAD

**Discussion by canseb** [strongswan/strongswan#1650] [Q&A]
Hello,

Strongswan Android client is up and running, but performance is much less than android native client. Using adb shell client, I can see that my CPU ( I have a Pixel 6a phone) support aes hardware acceleration. The command line top also show that native client doesn't use much CPU (indicating that it is using hardware capabilities), but strongswan use much more CPU (seams it is software crypto)

Do you know if it's possible to enable Hardware crypto capability for android client?  

Thank you!

**Comment by tobiasbrunner** [comment]
> The command line top also show that native client doesn't use much CPU

It also relies on the IPsec implementation in the LInux kernel and does not implement ESP in userland like the app does.

> Do you know if it's possible to enable Hardware crypto capability for android client?

Not sure what OpenSSL supports in that regard or how exactly it's enabled. But removing `no-asm` from the configure options [here](https://github.com/strongswan/strongswan/blob/5284cecddc4dbbbb0772313190af3aa690278123/src/frontends/android/openssl/compile.sh#L43) might do the trick.

**Comment by canseb** [comment]
Hello,

Thank you for your advice. I've recompiled and installed my home-compiled strongswan without no-asm option for openssl.
Speedtest application show there is some improvements. Was downloading at ~100Mbit. Without no-asm, got 20 to 60% improvement.

It seams it did something worthful, but didn't had time to do rigorouts tests.

Compiled vith openssl 3.2 and strongswan based on 5.9 release commit.
Arch is arm64-v8a and cpuinfo show that my cpu support aes
